### ISSUE TITLE
Feature request: add checkpoint auto save and resume training support

### ISSUE THREAD

**Issue by httpsVishu** [gml16/rl-medical#25]
while working with training in this repo, i noticed that there is not really a way to resume training if it gets interrupted

see right now, if training stops midway (may be due to a crash or manual stop), the progress is essentially lost and training has to be started again from scratch, given that training here can be quite long (especially for medical rl setups), this can be a bit limiting 

i was thinking it might be useful to add some basic checkpointing support like we can save the model state every n steps/episodes and we can add a flag like `--resume path/to/checkpoint`, with its help we can restore the model (and optimizer/scheduler if applicable) from that checkpoint

no doubt, this will add slightly more complexity in the training loop and additional disk usage depending on how frequently checkpoints are saved but all of these are manageable and this feature would really make experiments easier to manage and align more with typical ml workflows

not sure if something like this would be beneficial at this stage as this repo is half a decade old but thought it might be worth discussing for robustness and usability

**Comment by gml16** [comment]
That seems like a great idea. This can replace the `--load` argument, adding as you say optim/scheduler states. Very happy to merge such a PR if you’re interested in implementing it

**Comment by httpsVishu** [comment]
awesome, happy to work on this, will open a PR soon
